packages/create-subscription/src/__tests__/createSubscription-test.internal.js

```diff
@@ -0,0 +1,420 @@
+      expect(ReactNoop.flush()).toEqual([123]);
+    });
+  });
+
+  it('should unsubscribe from old subscribables and subscribe to new subscribables when props change', () => {
+    const Subscription = createSubscription({
+      getValue: source => source.getValue(),
+      subscribe: (source, callback) => {
+        const subscription = source.subscribe(callback);
+        return () => subscription.unsubscribe();
+      },
+    });
+
+    function render(value = 'default') {
+      ReactNoop.yield(value);
+      return null;
+    }
+
+    const observableA = createBehaviorSubject('a-0');
+    const observableB = createBehaviorSubject('b-0');
+
+    ReactNoop.render(
+      <Subscription source={observableA}>{render}</Subscription>,
+    );
+
+    // Updates while subscribed should re-render the child component
+    expect(ReactNoop.flush()).toEqual(['a-0']);
+
+    // Unsetting the subscriber prop should reset subscribed values
+    ReactNoop.render(
+      <Subscription source={observableB}>{render}</Subscription>,
+    );
+    expect(ReactNoop.flush()).toEqual(['b-0']);
+
+    // Updates to the old subscribable should not re-render the child component
+    observableA.next('a-1');
+    expect(ReactNoop.flush()).toEqual([]);
+
+    // Updates to the bew subscribable should re-render the child component
+    observableB.next('b-1');
+    expect(ReactNoop.flush()).toEqual(['b-1']);
+  });
+
+  it('should ignore values emitted by a new subscribable until the commit phase', () => {
+    let parentInstance;
+
+    function Child({value}) {
+      ReactNoop.yield('Child: ' + value);
+      return null;
+    }
+
+    const Subscription = createSubscription({
+      getValue: source => source.getValue(),
+      subscribe: (source, callback) => {
+        const subscription = source.subscribe(callback);
+        return () => subscription.unsubscribe();
+      },
+    });
+
+    class Parent extends React.Component {
+      state = {};
+
+      static getDerivedStateFromProps(nextProps, prevState) {
+        if (nextProps.observed !== prevState.observed) {
+          return {
+            observed: nextProps.observed,
+          };
+        }
+
+        return null;
+      }
+
+      render() {
+        parentInstance = this;
+
+        return (
+          <Subscription source={this.state.observed}>
+            {(value = 'default') => {
+              ReactNoop.yield('Subscriber: ' + value);
+              return <Child value={value} />;
+            }}
+          </Subscription>
+        );
+      }
+    }
+
+    const observableA = createBehaviorSubject('a-0');
+    const observableB = createBehaviorSubject('b-0');
+
+    ReactNoop.render(<Parent observed={observableA} />);
+    expect(ReactNoop.flush()).toEqual(['Subscriber: a-0', 'Child: a-0']);
+
+    // Start React update, but don't finish
+    ReactNoop.render(<Parent observed={observableB} />);
+    ReactNoop.flushThrough(['Subscriber: b-0']);
```

Comment:
Can we add some more assertions to prove that the component has not yet committed?

Reply:
Sure! I'll log and compare parent commit lifecycles.